orchestra/static/orchestra/project/js/controllers.js

```diff
@@ -0,0 +1,870 @@
+            'class': 'crosshair',
+            'stroke': 'rgb(0, 121, 191)',
+            'x1': 0,
+            'x2': 0,
+            'y1': 0,
+          });
+
+        this.draw = function() {
+          this.svg
+            .attr({
+              'height': vis.utils.getSvgHeight(),
+              'width': 1 // vis.utils.getSvgWidth()
+            })
+          this.line.attr({
+            'y2': function() {return vis.utils.getSvgHeight()}
+          })
+        }
+
+        this.show = function() {
+          this.draw();
+          this.svg.attr('opacity', 1);
+        }
+        this.hide = function() {
+          this.svg.attr('opacity', 0);
+        }
+        this.move = function(datetime) {
+          this.svg.style('left', (vis.options.marginLeft + vis.timeScale(datetime)) + 'px');
+          // this.line.attr('transform', vis.utils.translateString(vis.timeScale(datetime), 0))
+        }
+        this.reset = function() {
+          this.svg.attr('opacity', 0);
+          this.move(vis.timeScale.domain()[0]);
+        }
+      }
+
+      function actions() {
+        var actions = this;
+
+        actions.createSubsequentTasks = function() {
+          orchestraApi.createSubsequentTasks(vis.projectId)
+            .then(function() {vis.update()},
+                  function() {alert('Could not create subsequent tasks.')});
+        }
+
+        actions.skipTask = function(task) {
+          if (!confirm('Are you sure you want to skip this task and mark it as ' +
+                      'complete? All sorts of crazy things might happen.')) {
+            return;
+          }
+          orchestraApi.skipTask(task)
+            .then(function() {vis.update()},
+                  function() {alert('Error skipping task.')});
+        }
+
+        actions.addNewAssignment = function(task, inputEl) {
+          orchestraApi.addNewAssignment(task, inputEl.node().value)
+            .then(function() {vis.update()},
+                  function() {
+                    inputEl.node().value = '';
+                    inputEl.node().blur();
+                    alert('Error assigning task.');
+                  })
+            .finally(function() {vis.addingAssignment = false})
+        }
+
+        actions.reassign = function(assignment, inputEl) {
+          if (assignment.reassigning) {
+            return;
+          }
+          assignment.reassigning = true
+          orchestraApi.reassign(assignment, inputEl.node().value)
+            .then(function() {
+              assignment.worker = inputEl.node().value;
+              vis.assignments.draw();
+            }, function() {
+              inputEl.node().blur();
+              alert('Error reassigning worker.')
+            })
+            .finally(function() {
+              inputEl.node().blur();
+              inputEl.node().value = assignment.worker;
+              assignment.reassigning = false
+            })
+        }
+
+        actions.showProjectData = function() {
```

Comment:
sometimes the modal is in its own function, and sometimes it's baked into a larger function.  suggests we should have a modal service